lib/ansible/modules/cloud/scaleway/scaleway_sshkey.py

```diff
@@ -0,0 +1,263 @@
+        self.base_url = base_url
+
+    def _url_builder(self, path):
+        if path[0] == '/':
+            path = path[1:]
+        return '%s/%s' % (self.base_url, path)
+
+    def send(self, method, path, data=None, headers=None):
+        url = self._url_builder(path)
+        data = self.module.jsonify(data)
+        timeout = self.module.params['timeout']
+
+        if headers is not None:
+            self.headers.update(headers)
+
+        resp, info = fetch_url(self.module, url, data=data, headers=self.headers, method=method, timeout=timeout)
+
+        # Exceptions in fetch_url may result in a status -1, the ensures a proper error to the user in all cases
+        if info['status'] == -1:
+            self.module.fail_json(msg=info['msg'])
+
+        return Response(resp, info)
+
+    def get(self, path, data=None, headers=None):
+        return self.send('GET', path, data, headers)
+
+    def put(self, path, data=None, headers=None):
+        return self.send('PUT', path, data, headers)
+
+    def post(self, path, data=None, headers=None):
+        return self.send('POST', path, data, headers)
+
+    def delete(self, path, data=None, headers=None):
+        return self.send('DELETE', path, data, headers)
+
+    def patch(self, path, data=None, headers=None):
+        return self.send("PATCH", path, data, headers)
+
+    def update(self, path, data=None, headers=None):
+        return self.send("UPDATE", path, data, headers)
+
+
+def extract_present_sshkeys(raw_organization_dict):
+    ssh_key_list = raw_organization_dict["organizations"][0]["users"][0]["ssh_public_keys"]
+    ssh_key_lookup = [ssh_key["key"] for ssh_key in ssh_key_list]
+    return ssh_key_lookup
+
+
+def extract_user_id(raw_organization_dict):
+    return raw_organization_dict["organizations"][0]["users"][0]["id"]
+
+
+def sshkey_user_patch(ssh_lookup):
+    ssh_list = {"ssh_public_keys": [{"key": key}
+                                    for key in ssh_lookup]}
+    return ssh_list
+
+
+def core(module):
+    api_token = module.params['oauth_token']
+    ssh_pub_key = module.params['ssh_pub_key']
+    state = module.params["state"]
+    account_api = AccountAPI(module,
+                             headers={'X-Auth-Token': api_token,
+                                      'Content-type': 'application/json'},
+                             base_url=module.params["base_url"])
+    response = account_api.get('organizations')
+
+    status_code = response.status_code
+    organization_json = response.json
+
+    if status_code not in (200, 404):
+        module.fail_json(msg='Error getting ssh key [{0}: {1}]'.format(
+            status_code, response.json['message']))
+
+    user_id = extract_user_id(organization_json)
+    present_sshkeys = []
+    try:
+        present_sshkeys = extract_present_sshkeys(organization_json)
+    except (KeyError, IndexError) as e:
+        module.fail_json(changed=False, data="Error while extracting present SSH keys from API")
```

Comment:
wouldn't hurt to include some details about the error here. The original exception error message for example.

But thats not a blocking suggestion.